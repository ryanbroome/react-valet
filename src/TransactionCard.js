import React from "react";
import { Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardSubtitle, ListGroup, ListGroupItem, CardText } from "reactstrap";
import TransactionList from "./TransactionList";

const TransactionCard = ({ transaction, checkout }) => {
  return (
    // <div>
    <Card
      color="light"
      body
      className="text-center">
      <CardBody>
        <CardSubtitle>Ticket: {transaction.ticketNum}</CardSubtitle>

        <ListGroup flush>
          <ListGroupItem>Mobile: {transaction.mobile}</ListGroupItem>
          <ListGroupItem>
            <Link to={`/transactions/id/${transaction.transactionId}`}>Detail </Link>
          </ListGroupItem>
          <ListGroupItem>Color: {transaction.color}</ListGroupItem>
          <ListGroupItem>Make: {transaction.make}</ListGroupItem>
        </ListGroup>

        <ListGroup>
          <ListGroupItem>
            Parked By Valet: {transaction.firstName} {transaction.lastName}
          </ListGroupItem>
          <ListGroupItem>In Time: {transaction.checkIn}</ListGroupItem>
          <ListGroupItem>Damages: {transaction.damages}</ListGroupItem>
          <ListGroupItem>Notes: {transaction.notes}</ListGroupItem>
          <button
            onClick={() => {
              checkout(transaction.username, transaction.vehicleId);
            }}>
            CHECKOUT
          </button>
        </ListGroup>
        <ListGroup>
          {/* <ListGroupItem>Site: {transaction.sitename}</ListGroupItem>
          <ListGroupItem>Site ID: {transaction.locationId}</ListGroupItem>
          <ListGroupItem>
            <b>Transaction Info</b>
          </ListGroupItem> */}
          {/* <ListGroupItem>Trans: {transaction.transactionId}</ListGroupItem> */}
          {/* <ListGroupItem>Ticket:{transaction.ticketNum}</ListGroupItem> */}
          {/* <ListGroupItem>Trans Time: {transaction.transactionTime}</ListGroupItem> */}
          {/* <ListGroupItem>
            <b>Customer Vehicle Info</b>
          </ListGroupItem> */}
          {/* <ListGroupItem>Vehicle ID: {transaction.vehicleId}</ListGroupItem> */}
          {/* <ListGroupItem>Status: {transaction.vehicleStatus}</ListGroupItem> */}
          {/* <ListGroupItem>Out Time: {transaction.checkOut}</ListGroupItem> */}
          {/* <b>Valet Info</b> */}
          {/* </ListGroupItem> */}
          {/* <ListGroupItem>Valet ID: {transaction.userId}</ListGroupItem> */}
          {/* <ListGroupItem>Last: {transaction.lastName}</ListGroupItem> */}
          {/* <ListGroupItem>Phone: {transaction.phone}</ListGroupItem> */}
          {/* <ListGroupItem>Email: {transaction.email}</ListGroupItem> */}
          {/* <ListGroupItem>TotalParked: {transaction.totalParked > 0 ? transaction.totalParked : "None"}</ListGroupItem> */}
          {/* <ListGroupItem>Admin: {transaction.isAdmin ? "Yes" : "No"}</ListGroupItem> */}
        </ListGroup>
      </CardBody>
    </Card>
    // </div>
  );
};

export default TransactionCard;
