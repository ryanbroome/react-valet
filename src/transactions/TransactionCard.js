import React from "react";
import { Link } from "react-router-dom";
import { Card, CardBody, CardSubtitle, ListGroup, ListGroupItem, Button } from "reactstrap";

const TransactionCard = ({ transaction, checkout }) => {
  return (
    <Card
      color="light"
      body
      className="text-center">
      <CardBody>
        <CardSubtitle>Ticket: {transaction.ticketNum}</CardSubtitle>

        <ListGroup flush>
          <ListGroupItem>Mobile: {transaction.mobile}</ListGroupItem>
          <ListGroupItem>
            <Link to={`/transactions/id/${transaction.transactionId}`}>Detail </Link>
          </ListGroupItem>
        </ListGroup>
        <ListGroup></ListGroup>
        <ListGroup>
          <Button
            color="danger"
            onClick={() => {
              checkout(transaction.username, transaction.vehicleId);
            }}>
            CHECKOUT
          </Button>
        </ListGroup>
      </CardBody>
    </Card>
  );
};

export default TransactionCard;
