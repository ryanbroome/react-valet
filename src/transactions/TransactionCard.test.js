import React from "react";
import { render } from "@testing-library/react";
import TransactionCard from "./TransactionCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const transaction = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const { asFragment } = render(
    <MemoryRouter>
      <TransactionCard transaction={transaction} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
