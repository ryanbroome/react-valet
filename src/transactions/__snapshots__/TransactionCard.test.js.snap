// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="text-center card card-body bg-light"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-subtitle"
      >
        Ticket: 1
      </h6>
      <ul
        class="list-group list-group-flush"
      >
        <li
          class="list-group-item"
        >
          Mobile: 5553334444
        </li>
        <li
          class="list-group-item"
        >
          <a
            href="/transactions/id/1"
          >
            Detail 
          </a>
        </li>
      </ul>
      <ul
        class="list-group"
      />
      <ul
        class="list-group"
      >
        <button
          class="btn btn-danger"
          type="button"
        >
          CHECKOUT
        </button>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
